source/common/upstream/eds.h

```diff
@@ -82,6 +97,12 @@ class EdsClusterImpl : public BaseDynamicClusterImpl, Config::SubscriptionCallba
   Event::TimerPtr assignment_timeout_;
   ProtobufMessage::ValidationVisitor& validation_visitor_;
   InitializePhase initialize_phase_;
+
+  Config::SubscriptionFactory& subscription_factory_;
+  Upstream::ClusterManager& cm_;
+  EgdsApiPtr egds_api_;
+  EgdsClusterMapperPtr egds_cluster_mapper_;
+  EndpointGroupsManagerPtr endpoint_group_manager_;
```

Comment:
These members plus the subscription factory  and cluster manager are introduced to tackle the endpoint_groups but add very little value to the existing `EdsClusterImpl`. A strong signal is that these are rarely used except the inialization.
My impression is that these can be extracted.